lib/versioned/test.js

```diff
@@ -42,6 +43,29 @@ function Test(directory, pkgVersions, opts = {}) {
   const pkg = require(path.join(directory, 'package'))
   const dirname = path.basename(directory)
 
+  // compare test files included in specification
+  // against files that actually exist.
+  // warn or throw if files which exist are not specified.
+  const uniqSoughtFiles = new Set()
+  for (const test of pkg.tests) {
+    for (const file of test.files) {
+      uniqSoughtFiles.add(file)
+    }
+  }
+  const allFiles = fs.readdirSync(directory).filter((name) => {
+    return /.*\.js$/.test(name)
```

Comment:
Interesting so after updating this regex to 
```
  1 diff --git a/lib/versioned/test.js b/lib/versioned/test.js
  2 index fc1339e..4f6ec0c 100644
  3 --- a/lib/versioned/test.js
  4 +++ b/lib/versioned/test.js
  5 @@ -53,16 +53,16 @@ function Test(directory, pkgVersions, opts = {}) {
  6      }
  7    }
  8    const allFiles = fs.readdirSync(directory).filter((name) => {
  9 -    return /.*\.js$/.test(name)
 10 +    return /.*(tap|test)\.js$/.test(name)
 11    })
```

It found issues:

```
/Users/revans/code/node-test-utilities/lib/versioned/test.js:62
        throw new Error(errMsg)
        ^

Error: Found test with-express.tap.js which is not being run! Did you forget to specify it?
    at new Test (/Users/revans/code/node-test-utilities/lib/versioned/test.js:62:15)
    at /Users/revans/code/node-test-utilities/lib/versioned/suite.js:152:22
    at Array.map (<anonymous>)
    at Suite._runTests (/Users/revans/code/node-test-utilities/lib/versioned/suite.js:152:6)
    at nextTask (/Users/revans/code/node-test-utilities/node_modules/async/dist/async.js:5324:14)
    at next (/Users/revans/code/node-test-utilities/node_modules/async/dist/async.js:5331:9)
    at /Users/revans/code/node-test-utilities/node_modules/async/dist/async.js:969:16
    at loadHandler (/Users/revans/code/node-test-utilities/lib/versioned/suite.js:92:7)
    at /Users/revans/code/node-test-utilities/node_modules/async/dist/async.js:1140:9
    at /Users/revans/code/node-test-utilities/node_modules/async/dist/async.js:473:16
```

Reply:
It correctly found issues! Hooray! As this file is indeed not included: https://github.com/newrelic/node-newrelic/blob/main/test/versioned/restify/restify-pre-7/package.json#L16-L21

Reply:
Would it be appropriate to log the error and then exit the process, rather than just throwing the error? I say this because the stacktrace leading back to this logic will not be useful to a user experiencing the issue.

Reply:
> Would it be appropriate to log the error and then exit the process, rather than just throwing the error? I say this because the stacktrace leading back to this logic will not be useful to a user experiencing the issue.

Yea I don't think a stack trace is helpful here at all.  We just need to know at the end of a run if there are any files that were never executed

Reply:
Great! Here's what it looks like now:

```
 ✖ restify-post-7/restify.tap.js run 3 of 105): failure
 • restify-pre-7/capture-params.tap.js run 1 of 42): running
===============================================================
Found 1 test file outside of those specified:

- /Users/dthayer/code/node-newrelic/test/versioned/restify/restify-pre-7/with-express.tap.js

Include them in a test group's package.json or delete them.
```